esp/public/media/scripts/onsite/ajax_status.js

```diff
@@ -725,8 +731,23 @@ function toggle_categories() {
 
 }
 
+function extract_category_id(element_id) 
+{
+    return parseInt(element_id.split("_")[2]);
+}
+
 function render_category_options()
 {
+    var checkedCategories = [];
+    if(settings.is_initial) {
+        checkedCategories = settings.categories_to_display.slice();
+        settings.is_initial = false;
+    } else {
+        $j('#category_list input:checked').each(function() {
+            checkedCategories.push(extract_category_id($j(this).attr('id')));
+        });
+    }
+    
```

Comment:
I don't quite understand what the intent of all this code is -- why are we clearing the `#category_list` and re-rendering it again at all?